codemark/firebase/database.py

```diff
@@ -81,28 +81,34 @@ def signout(self):
         except Exception as e:
             print_error(e)
 
-    def register(self, email, password):
-        '''
+    def register(self, email, password, user_type, user_config, user_type_config):
+        """
         Registers a user to the firebase realtime database 
         :param email: Email
         :param password: Password
-        '''
+        :param user_type: Whether the user is a 'student' or a 'professor'
+        :param user_config: Dictionary containing the details of the user
+            Keys: 'email', 'user_type', 'name', 'college', 'student_id'
+        :param user_type_config: Dictionary containing the details specific to a user_type
+            Keys (student): 'batch_id', 'batch_semester', 'college_name', 'email', 'field_of_study', 'name', 'passing_year', 'student_id'
+            Keys (professor): 'college_name', 'name', 'professor_id'
+        """
         try:
             user = pauth.create_user_with_email_and_password(email, password)
+            res = self.pushData('users', user_config)
```

Comment:
> user = pauth.create_user_with_email_and_password(email, password)

As discussed in previous review, I will recommend using firebase_admin API here, (actually pyrebase is an third party library and is having some dependencies on very old versions, so kinda that's the whole reason, sooner or later we want to minimize or dependencies on that)

In case you're having trouble implementing it, then do ping me out, I can surely help you with that in part/full.

Reply:
The pushData function does use the firebase_admin api right at L143?

Reply:
Yes, indirectly!